eclipse/plugins/org.switchyard.tools.ui/src/org/switchyard/tools/ui/common/SwitchYardSettingsGroup.java

```diff
@@ -805,7 +805,15 @@ private void populateRuntimeVersionsList(IProgressMonitor monitor) {
         }
         // add default version
         final ArtifactVersion defaultVersion = parseVersion(NewSwitchYardProjectWizard.DEFAULT_RUNTIME_VERSION);
-        _availableVersions.add(defaultVersion);
+        if (!_availableVersions.contains(defaultVersion)) {
+            _availableVersions.add(defaultVersion);
+        }
+
+        // for SWITCHYARD-2959 - add previous version as well 
```

Comment:
As for popping this out to another method, I'm good with that - it holds with the principles of "Building Maintainable Software" pretty closely.